### ISSUE TITLE
Add `aria-sort` to Datagrid and improve screenreader experience

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#3247]
> There are a few screenreader issues when testing with voiceover setting on Mac. 
> 
> 2. Aria sort
> - Need to check markup of the table - should be using `aria-sort` so that the screenreader announces to the user that the table is sortable 
> 
> Current: ![Screenshot 2023-07-02 at 11 27 10 AM](https://github.com/carbon-design-system/ibm-products/assets/21047571/af21f712-32e8-4ce6-a585-437705397c24)
> 
> Current: ![Screenshot 2023-07-02 at 11 36 49 AM](https://github.com/carbon-design-system/ibm-products/assets/21047571/3c79bd1a-9635-4c12-a50a-a833608cc590)
> 
> Correct example from https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced:
>  ![Screenshot 2023-07-02 at 12 03 37 PM](https://github.com/carbon-design-system/ibm-products/assets/21047571/2978950a-c8f0-43d4-b1d8-868e0f5d18db)
> 
> 3. Reading first row after clicking sort button 
> - When clicking CNTRL+Option+space to click the sort button when using screenreader, the screenreader reads out the first table row, without any context.  It shouldn't be reading any table content. 
> ![Screenshot 2023-07-02 at 12 45 50 PM](https://github.com/carbon-design-system/ibm-products/assets/21047571/a7e7a0fc-9eba-4bff-ac80-a0c9bd41b5b1)

_Originally posted by @kristastarr in https://github.com/carbon-design-system/ibm-products/issues/3118#issuecomment-1616741501_
            
```[tasklist]
### Tasks
- [ ] Add `aria-sort` so that the screenreader announces to user that the table is sortable
- [ ] Prevent reading of first row when sorting via screenreader
```
